modules/project/variables-metrics-alerts.tf

```diff
@@ -0,0 +1,153 @@
+        renotify_interval          = optional(string)
+      }))
+    }))
+    conditions = optional(object({
+      display_name = optional(string)
+      condition_matched_log = optional(object({
+        filter           = optional(string)
+        label_extractors = optional(map(string))
+      }))
+      condition_monitoring_query_language = optional(object({
+        query    = optional(string)
+        duration = optional(string)
+        trigger = optional(object({
+          count   = optional(number)
+          percent = optional(number)
+        }))
+        evaluation_missing_data = optional(string)
+      }))
+      condition_threshold = optional(object({
+        threshold_value    = optional(number)
+        denominator_filter = optional(string)
+        denominator_aggregations = optional(object({
+          per_series_aligner   = optional(string)
+          group_by_fields      = optional(list(string))
+          cross_series_reducer = optional(string)
+          alignment_period     = optional(string)
+        }))
+        forecast_options = optional(object({
+          forecast_horizon = optional(string)
+        }))
+        comparison              = optional(string)
+        duration                = optional(string)
+        filter                  = optional(string)
+        evaluation_missing_data = optional(string)
+        resource_type           = optional(string)
+        trigger = optional(object({
+          count   = optional(number)
+          percent = optional(number)
+        }))
+        aggregations = optional(object({
+          per_series_aligner   = optional(string)
+          group_by_fields      = optional(list(string))
+          cross_series_reducer = optional(string)
+          alignment_period     = optional(string)
+        }))
+      }))
+      condition_absent = optional(object({
+        filter = optional(string)
+        aggregations = optional(object({
+          per_series_aligner   = optional(string)
+          group_by_fields      = optional(list(string))
+          cross_series_reducer = optional(string)
+          alignment_period     = optional(string)
+        }))
+        trigger = optional(object({
+          count   = optional(number)
+          percent = optional(number)
+        }))
+        duration = optional(string)
+      }))
+
+    }))
+  }))
+  nullable = false
+  default  = {}
+}
+
+variable "logging_metrics" {
+  description = "Logging metrics alerts configuration."
+  type = map(object({
+    name             = optional(string)
+    description      = optional(string)
+    filter           = optional(string)
+    bucket_name      = optional(string)
+    value_extractor  = optional(string)
+    label_extractors = optional(map(string))
+    disabled         = optional(bool)
+    metric_descriptor = optional(map(object({
+      metric_kind = optional(string)
+      value_type  = optional(string)
+      labels = list(object({
+        key         = optional(string)
```

Comment:
key is required

Reply:
Should `labels` be a map instead? It would also prevent duplicate keys.